happi/client.py

```diff
@@ -378,8 +378,15 @@ def _get_search_results(self, items, *, wrap_cls=None):
         Return search results to the user, optionally wrapping with a class
         '''
         wrap_cls = wrap_cls or self._results_wrap_class
-        return [wrap_cls(client=self, device=self.find_device(**info))
-                for info in items]
+        results = []
+        for info in items:
+            try:
+                result = wrap_cls(client=self, device=self.find_device(**info))
+                results.append(result)
+            except Exception:
+                logger.warning(f'Entry for {info["name"]} is malformed. '
```

Comment:
`f` strings in log messages are generally avoided, as they do their own string interpolation by way of the classic `%s` and can potentially raise exceptions which would be caught by the standard route. Not a big deal though here.

Reply:
Right, I forgot about this. Might as well switch it over now.